ir/type.cpp

```diff
@@ -614,11 +614,22 @@ AggregateType::AggregateType(string &&name, bool symbolic)
   }
 }
 
+AggregateType::AggregateType(string &&name, vector<Type*> &&vchildren,
```

Comment:
this design seems slightly complicated.
What do you think about explicitly adding padding fields to structs, so the API wouldn't change much. Each field would have a bool stating whether it's padding or not. Then we just need to detect padding when doing GEPs into structs, but the rest of the code (loads, stores, etc) wouldn't need any change. A padding field is a field like any other. Just with poison data I guess.

Reply:
The `vlayout` parameter exactly does what you've suggested. It takes (beginning offset, field size, padding). Detection of padding is done at utils.cpp . I think load/store still need change because they need to access elementwise (and respect endian of each element while preserving element order).